### PR TITLE
Description of forking, and how to do it

### LINKED ISSUE #421: Document PITR (Citus Cloud)
Relevant blog-post: WIP: https://docs.google.com/document/d/1NyEaB6KGQobt093w2bv8oEKGuI0ss1WAWW-5GHuPHH0/edit?ts=595cfd1f

### PR DESCRIPTION
**Pull request by begriffs**
Fixes #421 

The referenced issue is about PITR, which is more general than forking. I have a draft on my local machine of the content for PITR, and this PR includes just the trimmed down forking instructions.

### CODE REVIEW

File: cloud/forking.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+`Log shipping <https://www.postgresql.org/docs/current/static/continuous-archiving.html#BACKUP-ARCHIVING-WAL>`_ in PostgreSQL calls a user-provided command that takes a WAL file and stores it somewhere. This command can be a simple ``cp`` that will copy the WAL file to a disk in the network or something more capable like `WAL-E <https://github.com/wal-e/wal-e>`_.
+
+Base backups and WAL archives are all that is needed to restore the database to some specific point in time. First we restore a base backup that was before the desired restoration time. Then we create a `recovery.conf <https://www.postgresql.org/docs/current/static/recovery-config.html>`_ file with the target time, the command to restore a requested WAL file from the archives, and other configuration options. The new PostgreSQL instances, upon entering recovery mode, will start playing WAL segments up to the target point. After the recovery instances reach the specified target, they will be available for use as a regular database.
```
Comment (sumedhpathak) [inline cloud/forking.rst]: This seems a bit detailed for Docs. I think a brief overview is sufficient. Could we also add a bit on potential use-cases for fork?

Comment (begriffs) [inline cloud/forking.rst]: After trimming things out of the bigger pitr document adapted from things Craig wrote, the result looked pretty small. I included the internals section for the extra padding, but I'll remove it and add a part elaborating potential uses. Agreed that the internals are at a way different level of detail from the rest of the cloud docs.

File: cloud/forking.rst [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+How to Fork a Formation
+-----------------------
+
+Citus Cloud makes forking easy. The control panel for each formation has a "Fork" tab. Go there and enter the name, region, and node sizing information for the destination cluster.
+
+.. image:: ../images/cloud-fork.png
+
+Shortly after you click "Fork Formation," the new formation will appear in the Cloud console. It runs on separate hardware and your database can connect to it in the :ref:`usual way <connection>`.
+
+When is it Useful
```
Comment (sumedhpathak) [inline cloud/forking.rst]: I'm also OK to have a more abbreviated version of the "under the hood" section previously.

Comment (begriffs) [inline cloud/forking.rst]: Cool, added one.

**Pull request merged**